### ISSUE TITLE
FZF-Lua

### ISSUE THREAD

**Issue by mattiasbonte** [EvWilson/spelunk.nvim#13]
I love this plugin! Also love fzf-lua. Would it be possible to add support for it?

**Comment by EvWilson** [comment]
Hey @mattiasbonte! I think I could be convinced to add support for fzf-lua in addition to the telescope support, though I'm moreso considering supplying API hooks and some example docs to allow others to build their own experiences. This would mostly be to keep the default experience of the plugin more tightly scoped.

Working through some other enhancements right now, but I think I should be able to whip something together by the start of next week.

**Comment by EvWilson** [comment]
Hey @mattiasbonte, I experimented with using `fzf-lua` to emulate what we're doing with telescope right now, but discovered that there's no way to pass non-string structures around through the available pickers. Unfortunately, it would lead to a lot of nasty string wrangling to extract mark information from the full mark string.

Knowing all that now, I made some of the internal state available that I'm using right now for the telescope integration. It should be available in the above linked PR (https://github.com/EvWilson/spelunk.nvim/pull/16). Hopefully that should be enough in case you want to take a crack at it in your config, and I'd always be happy to add a custom setup to the FAQs samples if you get it working!
